rdkit/Chem/MolDb/Loader_orig.py

```diff
@@ -70,27 +70,33 @@ def ProcessMol(mol, typeConversions, globalProps, nDone, nameProp='_Name', nameC
         pD[pn] = typeConversions[colTyp][1](pv)
       else:
         pD[pn] = pv
-  else:
-    pD = {}
+
   if redraw:
     AllChem.Compute2DCoords(m)
   if not skipSmiles:
     row.append(Chem.MolToSmiles(mol, True))
-  row.append(DbModule.binaryHolder(mol.ToBinary()))
-  row.append(pD)
+  row.extend([DbModule.binaryHolder(mol.ToBinary()), pD])
   return row
 
 
 def ConvertRows(rows, globalProps, defaultVal, skipSmiles):
+  #  globalProps is a python dictionary
+  PropsSize: int = len(globalProps)
   for i, row in enumerate(rows):
-    newRow = [row[0], row[1]]
+    newRow = [0] * (PropsSize + 3 + int(not skipSmiles))
+    newRow[0] = row[0]
+    newRow[1] = row[1]
+    
     pD = row[-1]
-    for pn in globalProps:
-      pv = pD.get(pn, defaultVal)
-      newRow.append(pv)
-    newRow.append(row[2])
+    for counter, pn in enumerate(globalProps): # pn is the key
+      newRow[2 + counter] = pD.get(pn, defaultVal)
+
     if not skipSmiles:
-      newRow.append(row[3])
+      newRow[-2] = row[2]
+      newRow[-1] = row[3]
+    else:
+      newRow[-1] = row[2]
```

Comment:
If we keep this rewrite, reverse the if statement to have a simpler condition (no double negative - not not adding):
```
    if skipSmiles:
      newRow[-1] = row[2]
    else:
      newRow[-2] = row[2]
      newRow[-1] = row[3]
```

Reply:
OK, I got your idea! I would fix it in the next patch. Thank you for your review.